apps/backend/src/modules/groups/user-groups-to-permissions.model.ts

```diff
@@ -0,0 +1,20 @@
+import {
+	AbstractModel,
+	DatabaseTableName,
+} from "~/libs/modules/database/database.js";
+
+class UserGroupsToPermissionsModel extends AbstractModel {
```

Comment:
I believe we can get rid of these "through" models, like something to something. We can define relationMappings in Groups model. Like:

```javascript
	permissions: {
				join: {
					from: '',
					through: { // groups to permissions table relation here
						from: ''
						to: '',
					to: '',
				},
				modelClass: PermissionModel,
				relation: Model.ManyToManyRelation,
			},
```

And then we can also simplify the logic in repositories and services to have something like this:

```javascript
// on service level payload will look like this
const group = {
  // ...props to create a group
  permissions: permissionIds,
  users: userIds,
}

// on repository level
const createdGroup = await this.groupModel
			.query()
			.insertGraph(group.toNewObject(), { relate: true })
			.returning("*")
			.withGraphJoined(RelationName.PERMISSIONS)
             .withGraphJoined(RelationName.USERS)
			.execute();
```
ps. the above is just an example, it may look different